zerver/openapi/openapi.py

```diff
@@ -56,15 +108,34 @@ def spec(self) -> Dict[str, Any]:
         # earlier version than the current one
         if self.last_update != last_modified:
             self.reload()
-        assert(self.data)
+        assert(len(self.data))
         return self.data
 
+    def regex_keys(self) -> Dict[str, str]:
+        """Reload the OpenAPI file if it has been modified after the last time
+        it was read, and then return the parsed data.
+        """
+        last_modified = os.path.getmtime(self.path)
+        # Using != rather than < to cover the corner case of users placing an
+        # earlier version than the current one
+        if self.last_update != last_modified:
+            self.reload()
+        assert(len(self.regex_dict))
+        return self.regex_dict
+
+
 class SchemaError(Exception):
     pass
 
 openapi_spec = OpenAPISpec(OPENAPI_SPEC_PATH)
 
 def get_schema(endpoint: str, method: str, response: str) -> Dict[str, Any]:
+    if len(response) == 3 and ('oneOf' in (openapi_spec.spec())['paths'][endpoint]
+                               [method.lower()]['responses'][response]['content']
+                               ['application/json']['schema']):
+        # Currently at places where multiple schemas are defined they only
+        # differ in example so either can be used.
+        response += '_0'
```

Comment:
This is super hacky, but that's more the fault of the structure for how we record multiple possible responses right now.  

I think the ideal logic would just try matching against each of the response types and accept any successful match, but let's keep that change out of scope for this PR.